### PR TITLE
Fix some browse dataset script bugs and draw textdet gt instance with ignore flags

### PR DESCRIPTION
**Pull request by KevinNuNu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

1.textdet可视化gt_instance 增加根据ignore属性情况来可视化的功能
2.明确recog_lmdb_dataset在准备数据时属性值含义，将img_key内容赋给results['img_path']会导致后续的一些奇怪的报错
3.LoadImageFromNDArray函数不再强制赋值results['img_path']=None，同2会导致后续一些奇怪的报错
4.修复make_grid函数pipeline模式下显示ori_shapes数值错误的bug
5.删除InspectCompose 记录原始数据部分，修复后续找不到'data_sample'关键字的bug
6.修复auto+original模式下默认只使用LoadImageFromFile的图片读取方式的bug，优化为优先选用自定义图片读取方式
7.修复可视化test数据集时，由于test_pipeline的LoadOCRAnnotation位于倒数第二步gt_instance都为原图尺度，导致在缩放后的图像上画上了原图尺度的标注的bug
8.修复textrecog读取灰度图片时产生的bug

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmocr/blob/main/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmocr/blob/main/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmocr/datasets/recog_lmdb_dataset.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def parse_data_info(self,
         """
         data_info = {}
         img_key, text = raw_anno_info
-        data_info['img_path'] = img_key
+        data_info['img_key'] = img_key
```
Comment (Harold-lkk) [inline mmocr/datasets/recog_lmdb_dataset.py:129]: no need to rename

File: mmocr/visualization/textdet_visualizer.py [hunk 2]
```diff
@@ -142,6 +142,9 @@ def add_datasample(self,
         if data_sample is not None:
             if draw_gt and 'gt_instances' in data_sample:
                 gt_instances = data_sample.gt_instances
+                if gt_instances.get('ignored', None) is not None:
```
[the remark this answers is not in the document]
Comment (KevinNuNu) [reply mmocr/visualization/textdet_visualizer.py:155]: It's a good suggestion!

**Comment by Harold-lkk** [comment]
what is the problem when img_path is set in recog_lmdb_dataset?

**Comment by KevinNuNu** [comment]
> what is the problem when img_path is set in recog_lmdb_dataset?

Recog_lmdb_dataset里现在img_path这个属性实际赋的值时img_key，后续在调用results['img_path']的时候拿到的实际是img_key( like ‘image-0000001’)，例如这里就会报错，拿到的image-000001不是原文件的真实文件名。https://github.com/open-mmlab/mmocr/blob/27b6a68586b9a040678fe083bcf60662ae1b9261/tools/analysis_tools/browse_dataset.py#L380-L389

**Review by Harold-lkk — APPROVED** [review]

**Pull request merged**